org.eclipse.scanning.test/src/org/eclipse/scanning/test/malcolm/AbstractMalcolmTest.java

```diff
@@ -277,6 +282,7 @@ public void run() {
 	    return device;
 	}
 
+	@SuppressWarnings("squid:S2925")
 	protected synchronized void checkPauseResume(IMalcolmDevice device, long pauseTime, boolean ignoreReady) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)